lab4/provided/processor.cc

```diff
@@ -21,24 +19,30 @@ Processor::~Processor()
 }
 
 void
-Processor::sendRequest(Record &r)
+Processor::run()
 {
-    if (r.write) {
-        r.data = new uint8_t[r.size];
-        memset(r.data, (uint8_t)r.requestId, r.size);
-    }
+    createRecords();
+
+    if (trace.empty()) return;
```

Comment:
This should be an error case, no? Maybe `assert(!trace.empty())`.

Reply:
I specifically handle this not as an error case, so that the simulation will run and report "0 records run" if you load an empty txt in.